src/javascripts/Services/referencedByViewService.js

```diff
@@ -19,10 +19,46 @@ ngapp.service('referencedByViewService', function($timeout, layoutService, setti
         scope.onDrop = function() {
             let dragData = scope.$root.dragData;
             if (!dragData || dragData.source !== 'treeView') return;
-            let node = dragData.node,
-                path = node.element_type === xelib.etFile ? 'File Header' : '';
-            scope.record = xelib.GetElementEx(node.handle, path);
-            scope.syncWithLinkedViews(scope.record);
+            let node = dragData.node;
+            scope.open(xelib.GetElementEx(node.handle, ''));
+        };
+
+        scope.toggleSort = function(column) {
+            if (!column.canSort) return;
+            if (scope.sort.column !== column.label) {
+                scope.sort.column = column.label;
+                scope.sort.reverse = false;
+            } else {
+                scope.sort.reverse = !scope.sort.reverse;
+            }
+            scope.sortTree();
+        };
+
+        scope.sortTree = function() {
+            const colIndex = scope.allColumns.findIndex(column => {
+                return column.label === scope.sort.column;
+            });
+            if (colIndex === -1) return;
+            scope.tree.sort((a, b) => {
+                if (!a.has_data) scope.getNodeData(a);
```

Comment:
doing this in the comparator is unnecessarily slow.  I'd prefer a function call before this which gets data for all nodes, or even just getting data for all nodes as a part of `buildTree`.  If we do it in `scope.buildTree` we can just do the following:

```js
scope.tree.forEach(scope.getNodeData);
```

the end result is the same and we don't have to check every node's has_data property n * log(n)+ times.

Reply:
Good idea :)

Reply:
Furthermore, if we build the data all at once when the tree initializes (which we have to do if we want it to be sorted on the JS-side) there's no reason why we need to have `ng-init="node.has_data || getNodeData(node)"` in the HTML.